python/revng/tupletree/__init__.py

```diff
@@ -370,3 +389,323 @@ def init_check_kw_only(self, *args, **kwargs):
         original_init(self, *args, **kwargs)
 
     base_class.__init__ = init_check_kw_only  # type: ignore
+
+
+def construct_type(type_info: dict, data):
+    # MetaAddress comes already constructed so we jut return it
+    if type_info[type].__name__ == "MetaAddress":
```

Comment:
That is because in the `TypeHints` we use the `type` as a key, example:
```
TypeHints[Type] = {
    "Kind": {
        type: str,
        "possibleValues": TypeKind,
        "ctor": "enum",
        "optional": False,
        "isArray": False,
        "isAbstract": False,
    },
    "IsConst": {
        type: bool,
        "ctor": "native",
        "optional": True,
        "isArray": False,
        "isAbstract": False,
    },
}
```

Reply:
I thought it was intentional, will convert to `"type"`